### PR TITLE
Porting GeometryFeature to NTS

### PR DESCRIPTION
**Pull request by pauldendulk**
Making a start

### CODE REVIEW

**Review by inforithmics — APPROVED** [review]

**Comment by pauldendulk** [comment]
@inforithmics @janusw I have a problem with the github actions build. I made a seemingly small change that triggered it. I replaced the removed Mapsui.GeometryLayers with Mapsui.Nts. This succeeds on winBuild, but fails on macBuild and mauiBuild. I have now fixed the macBuild by removing the maui install (it was not used in the sln) and removing the 6.0.x build (5.0.x builds correctly). This should of course be reverted once this problem is solved. MAUI still fails with:

```console    
C:\\Users\\runneradmin\\AppData\\Local\\Microsoft\\dotnet\\sdk\\6.0.101\\Sdks\\Microsoft.NET.Sdk\\targets\\Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1004: Assets file 'D:\\a\\Mapsui\\Mapsui\\Mapsui.Nts\\obj\\project.assets.json' not found. Run a NuGet package restore to generate this file. [D:\\a\\Mapsui\\Mapsui\\Mapsui.Nts\\Mapsui.Nts.csproj]

```

Perhaps this problem is with the slns. I installed Visual Studio prerelease so I can build and run MAUI, no problem there. I would assume ```run: nuget restore Mapsui.Maui.sln``` should take take of the restore but perhaps there is something wrong with it (Mapsui.Nts has been added and I can run it on my Windows machine).

Any hints how to solve or investigate this problem are welcome.

**Comment by pauldendulk** [comment]
> Perhaps this problem is with the slns. 

I should have asked for help earlier, never underestimate the power of the rubber duck. I now see that Mapsui.Nts was not added in the maui sln, even though it did build correctly. It is present in the mac sln but there may be a problem with the guids.

File: Mapsui.UI.Forms/Objects/Polygon.cs [hunk 1]
```diff
@@ -77,12 +80,25 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
             switch (propertyName)
             {
                 case nameof(Positions):
-                    if (Feature.Geometry != null)
-                        ((Geometries.Polygon)Feature.Geometry).ExteriorRing = new LinearRing(Positions.Select(p => p.ToPoint()).ToList());
+                    var coordinates = Positions.Select(p => p.ToCoordinate()).ToList();
```
Comment (pauldendulk) [inline Mapsui.UI.Forms/Objects/Polygon.cs:83]: If Shells are added while Holes already exist they will be lost. Better to create everything from scratch every time.

File: NuSpec/Mapsui.ArcGIS.nuspec [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
```
Comment (pauldendulk) [inline NuSpec/Mapsui.ArcGIS.nuspec:2]: ArcGIS in a separate nuspec. It is an odd part of the code. Seems not to be used at all. It has a newtonsoftjson dependency that I can now remove from Mapsui.

File: Samples/Mapsui.Samples.Common/Maps/Demo/InfoLayersSample.cs [hunk 3]
```diff
@@ -101,40 +100,34 @@ private static GeometryFeature CreateLineFeature()
...
-                    {
-                        new Point(4000000, 5000000),
-                        new Point(4000000, 4000000),
-                        new Point(3000000, 4000000),
-                        new Point(3000000, 5000000),
-                        new Point(4000000, 5000000)
-                    }))
-                }
-            };
+            return new MultiPolygon(new[] {
```
Comment (pauldendulk) [inline Samples/Mapsui.Samples.Common/Maps/Demo/InfoLayersSample.cs:103]: A lot of rewriting went into the fact that you can not add coordinates to a NTS geometry.

File: Samples/Mapsui.Samples.Common/Maps/GeodanOfficesSample.cs [hunk 4]
```diff
@@ -2,16 +2,15 @@
...
 
 namespace Mapsui.Samples.Common.Maps
 {
     public class GeodanOfficesSample
     {
         public static MemoryLayer CreateLayer()
         {
-            var geodanAmsterdam = new Geometries.Point(122698, 483922);
-            var geodanDenBosch = new Geometries.Point(148949, 411446);
+            var geodanAmsterdam = new MPoint(122698, 483922);
```
Comment (pauldendulk) [inline Samples/Mapsui.Samples.Common/Maps/GeodanOfficesSample.cs:12]: Prefer MPoint if no other nts types are used.

File: Samples/Mapsui.Samples.Common/Maps/Special/WritableLayerSample.cs [hunk 5]
```diff
@@ -33,8 +31,7 @@ private void MapControlOnInfo(object sender, MapInfoEventArgs e)
 
             _writableLayer?.Add(new GeometryFeature
             {
-                Geometry =
-                new Point(e.MapInfo.WorldPosition.X, e.MapInfo.WorldPosition.Y)
+                Geometry = new Point(e.MapInfo.WorldPosition.X, e.MapInfo.WorldPosition.Y)
```
Comment (pauldendulk) [inline Samples/Mapsui.Samples.Common/Maps/Special/WritableLayerSample.cs:34]: The WriteableLayer is now specifically for NTS. Simply adding points without NTS should also be an option. We should not need a lot of custom code for that.

**Comment by pauldendulk** [comment]
There is still a problem with WFS. I ported it. Wanted to test it. Did not work. Looked at master. Did not work either. The error is: 

```
System.Exception: 'A service exception occured: Unknown namespace [p_bz-cadastre]'
```
Perhaps something changed in the service. Perhaps we need to add the namespace while parsing. I want to continue with the NTS without fixing this.

**Pull request merged**